BoardBank/MonoBank/SettingsViewModel.swift

```
@@ -0,0 +1,66 @@
+//
+//  SettingsViewModel.swift
+//  BoardBank
+//
+//  Created by Salwa Kisswani on 7/29/19.
+//  Copyright © 2019 Richard Neitzke. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+final class SettingsViewModel {
+
+    let bankManager = BankManager()
+    let defaultEmailURL = URL(string: "mailto:richardneitzke.rn+MonoBank@gmail.com")
+    let defaultWebPage = URL(string: "https://icons8.com")
+
+    func getCurrencySymbol() -> String {
+        return bankManager.currencySymbol
+    }
+
+    func getEnableSound() -> Bool {
+        return bankManager.soundsEnabled
+    }
+
+    func getDefaultBalance() -> String {
+        return String(bankManager.defaultBalance)
+    }
+
+    func getQuickAddAmount() -> String {
+        return String(bankManager.quickAddAmount)
+    }
+
+    func setCurrencySymbol(newSymbol: String) {
+        bankManager.currencySymbol = newSymbol
+    }
+
+    func setQuickAddAmount(newQuickAddAmount: String)  {
+        bankManager.quickAddAmount = Int32(newQuickAddAmount) ?? 200
+    }
+
+    func setDefaultBalance(newDefaultBalance: String) {
+        bankManager.defaultBalance = Int32(newDefaultBalance) ?? 1500
+    }
```

Maybe not now but possibly later you might want to refactor these two so that instead of putting in these default values if converting to an Int32 fails, instead inform the user that they've put in something invalid and asking them to either fix it or cancel their changes.

---

As a note as to why, there are perfectly reasonable cases for these setter functions to fail internally, such as when the strings are things like "Twenty" or "10t3"

---

I like it Chris. In the near future we may want to 'clean' the strings in multiple ways. For now, I think this suits our needs